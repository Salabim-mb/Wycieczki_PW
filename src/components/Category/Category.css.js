import styled from 'styled-components'
import Container from '@material-ui/core/Container';

const StyledContainer = styled(Container)`
    margin:0;
    padding: 0!important;
`;

export default StyledContainer