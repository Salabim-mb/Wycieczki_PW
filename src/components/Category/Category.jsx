import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Container from '@material-ui/core/Container';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { useParams } from "react-router-dom";


import Image from './components/Image'

import logo from '../../logo.svg'

const dummyData = [
    {
        title: 'XDD',
        date: '21.21.2002'
    },
    {
        title: 'XDD',
        date: '21.21.2002'
    },
    {
        title: 'XDD',
        date: '21.21.2002'
    },
]

const Category = () => {
    const { id } = useParams();

    const reciveData = () => {
        console.log(id)
    }

    return (
        <Container>
            <Image imgSrc={logo} imgAlt="obrazek" />
            {dummyData.map((data) => (
                <Card >
                    <CardContent>
                        <Typography color="textSecondary" gutterBottom>
                            {data.date}
                        </Typography>
                        <Typography variant="h5" component="h2">
                            {data.title}
                        </Typography>
                    </CardContent>
                    <CardActions>
                        <Button size="small" onClick={reciveData}>Learn More</Button>
                    </CardActions>
                </Card>
            ))}

        </Container>
    );
}
export default Category