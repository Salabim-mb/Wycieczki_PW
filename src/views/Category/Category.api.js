import paths from 'constants/api'


export const getPosts = async (params) => {
    const url = `${paths.BLOG_POSTS}?topic=${params.category}`;

    const headers = {
        'Content-Type': 'application/json',
    };

    const response = await fetch(url, { headers, method: 'GET' });

    if (response.status !== 200) {
        throw new Error('błąd');
    }

    return response.json();
};

export const getCategory = async (params) => {
    const url = `${paths.BLOG_TOPIC}${params.category}/`;

    const headers = {
        'Content-Type': 'application/json',
    };

    const response = await fetch(url, { headers, method: 'GET' });

    if (response.status !== 200) {
        throw new Error('błąd');
    }

    return response.json();
};