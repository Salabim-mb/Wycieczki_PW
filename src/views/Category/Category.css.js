import styled from 'styled-components'
import Container from '@material-ui/core/Container';

const StyledContainer = styled(Container)`
    margin:0 auto;
    padding: 0!important;
`;

const StyledPostDisplay = styled.div`
    padding: ${({ theme }) => theme.margin.m};
`;

export { StyledContainer, StyledPostDisplay }