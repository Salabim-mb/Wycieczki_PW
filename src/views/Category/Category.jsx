import React from 'react';
import { useQuery } from 'react-query'
import { useParams } from "react-router-dom";
import PostTile from 'components/PostTile'
import paths from 'constants/api'
import Typography from '@material-ui/core/Typography';
import Image from './components/Image'
import StyledContainer from './Category.css'


const Category = () => {
    const params = useParams();


    const getPosts = async () => {
        const url = `${paths.BLOG_POSTS}?topic=${params.category}`;

        const headers = {
            'Content-Type': 'application/json',
        };

        const response = await fetch(url, { headers, method: 'GET' });

        if (response.status !== 200) {
            throw new Error('błąd');
        }

        return response.json();
    };

    const getCategory = async () => {
        const url = `${paths.BLOG_TOPIC}${params.category}/`;

        const headers = {
            'Content-Type': 'application/json',
        };

        const response = await fetch(url, { headers, method: 'GET' });

        if (response.status !== 200) {
            throw new Error('błąd');
        }

        return response.json();
    };

    const posts = useQuery('posts', getPosts)
    const category = useQuery('category', getCategory)

    return (
        <StyledContainer>
            {category.isError ? <div>Nie udało się pobrać kategorii</div> : <>
                <Image imgSrc={`${paths.PLAIN}${category?.data?.cover_url}`} imgAlt={category?.data?.title} />
                <Typography variant="h2">{category?.data?.title}</Typography>
            </>}

            { posts.isError ? <div> Nie udało się pobrać postu </div> : posts?.data?.results?.map((data) => (
                <PostTile key={data.id} id={data.id} cover={data.cover} title={data.title} summary={data.summary} link={`/${params.category}/${data.id}`} />
            ))}
        </StyledContainer>
    );
}
export default Category