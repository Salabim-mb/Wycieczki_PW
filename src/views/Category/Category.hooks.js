import { useQuery } from 'react-query'
import { getCategory, getPosts } from './Category.api'

export const useQueryPosts = (params) => useQuery('posts', () => getPosts(params), { retry: false })
export const useQueryCategory = (params) => useQuery('category', () => getCategory(params), { retry: false })