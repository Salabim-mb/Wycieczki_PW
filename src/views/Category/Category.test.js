import React from 'react';
import Category from './Category'
import { BrowserRouter } from 'react-router-dom';
import { render, waitForElement } from '@testing-library/react';
// import logo from '../../logo.svg'


// jest.mock('../../components/PostTile', () => ({
//     PostTile: ({ cover, title, summary, link }) => <div><img src={cover} /><h2>{title}</h2><p>{summary}</p><a href={link} /></div>,
// }));


// jest.mock('./components/Image', () => ({
//     Image: ({ imgSrc, imgAlt }) => <img src={imgSrc} alt={imgAlt} />,
// }));

describe('Category - dashboard', () => {
    let failFetch = false;
    const data = [
        {
            id: 0,
            title: 'Matematyka',
            summary:
                'Proin ullamcorper vitae lorem eget tristique. Mauris urna nibh, vestibulum eget dolor ut, ornare scelerisque mauris. Sed semper hendrerit facilisis. ',
            link: '/blog/liceum/1',
            cover: '../../logo.svg',
        },
    ];

    beforeAll(() => {
        global.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve) => {
                if (failFetch) {
                    resolve({ status: 500 });
                }
                resolve({
                    status: 200,
                    json: () => Promise.resolve(data),
                });
            });
        });
    });

    beforeEach(() => {
        failFetch = false;
        jest.clearAllMocks();
    });

    it('should match snapshot', async () => {
        const { container, getByText } = render(
            <BrowserRouter>
                <Category />
            </BrowserRouter>
,
        );

        expect(window.fetch).toHaveBeenCalledTimes(1);
        await waitForElement(() => getByText('Matematyka'));

        expect(container).toMatchSnapshot();
    });

    // it('should show error message if fetch fails', async () => {
    //     failFetch = true;

    //     const { container, getByText } = render(
    //         <BrowserRouter>
    //             <Category />
    //         </BrowserRouter>,
    //     );

    //     expect(window.fetch).toHaveBeenCalledTimes(1);
    //     await waitForElement(() => getByText('Nie udało się pobrać postów.', { exact: false }));

    //     expect(container).toMatchSnapshot();
    // });
});
