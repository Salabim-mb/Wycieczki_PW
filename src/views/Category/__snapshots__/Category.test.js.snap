// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should match snapshot - data 1`] = `
<div>
  <div
    class="MuiContainer-root sc-eCssSg cnnlHQ MuiContainer-maxWidthLg"
  >
    <img
      alt="abc"
      class="sc-hKgILt gikrGG"
      src="https://wut-vtrips.herokuapp.com/link"
    />
    <h3
      class="MuiTypography-root MuiTypography-h3"
    >
      abc
    </h3>
  </div>
</div>
`;

exports[`Category should match snapshot - isError 1`] = `
<div>
  <div
    class="MuiContainer-root sc-eCssSg cnnlHQ MuiContainer-maxWidthLg"
  >
    <div>
       Nie udało się pobrać postu 
    </div>
  </div>
</div>
`;

exports[`Category should match snapshot - isLoading 1`] = `
<div>
  <div
    class="MuiContainer-root sc-eCssSg cnnlHQ MuiContainer-maxWidthLg"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
      style="width: 210px; height: 118px;"
    />
    <h3
      class="MuiTypography-root MuiTypography-h3"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
      style="width: 40px; height: 40px;"
    />
  </div>
</div>
`;
